### ISSUE TITLE
getifaddrs and freeifaddrs not in libc for illumos/solaris 10

### ISSUE THREAD

**Issue by sjorge** [pydron/ifaddr#23]
I'm having some issues on SmartOS (illumos) because getifaddrs and freeifaddrs are not in libc, but in libsocket.

Replacing the ctypes load to socket instead of c makes it work, not sure how to best dynamically select the other one on illumos.

I'm not sure how to only do it when on illumos or solaris 10 though.

**Comment by sjorge** [comment]
Something like this might work

```
try:
	retval = libc.func()
except AttributeError as e:
	if os.uname()[0] == 'SunOS':
		libc = ctypes.CDLL(ctypes.util.find_library("socket"), use_errno=True)
		retval = libc.func()
	else:
		raise e
```

**Comment by liv3010m** [comment]
Thanks!

**Comment by jstasiak** [comment]
I trust https://github.com/pydron/ifaddr/commit/0e469adedd20029d43a47e8273c7d01db1beffe0 (in master now) handles this.

**Comment by sjorge** [comment]
I think that’s good yeah.

**Comment by liv3010m** [comment]
Thanks to everybody involved on this.

**Comment by jstasiak** [comment]
The fix has been released in 0.1.7 just now.
